templates/feature_group/sdk_delete_post_request.go.tpl

```diff
@@ -0,0 +1,9 @@
+// If a delete failed, requeue on delete.
+if err == nil {
+        if foundResource, err := rm.sdkFind(ctx, r); err != ackerr.NotFound {
+                if isDeleteFailed(foundResource) {
+                        return requeueWaitWhileDeleteFailed
```

Comment:
why is the requeue on DeleteFailed status?

write out what we are trying to achieve in the PR description. It should help you understand